### PR TITLE
Update deployment scripts to work on Mac and supposed to work on Win …

### PR DESCRIPTION
**Pull request by adalrsjr1**
The deployment scripts for Docker weren't working due to the `--net=host`, this kind of networking doesn't work on Mac. To work correctly, Kruize should use a custom network. This pull request fixes the issue #89 by creating a custom network `kruize-network` in case it doesn't exist. Besides, all URLs related to Docker deployment were updated to use the container's name over `localhost` -- native Docker DNS resolution.

### CODE REVIEW

**Comment by adalrsjr1** [comment]
I didn't realize I removed .gitignore, my bad, it wasn't my intention.

File: scripts/docker-helpers.sh [hunk 1]
```diff
@@ -133,20 +135,29 @@ function docker_prereq() {
...
 	fi
+
+  NET_NAME=`docker network ls -f "name=${NETWORK}" --format {{.Name}} | tail -n 1`
+
+  if [[ -z $NET_NAME ]];  then
+    echo "Creating Kruize network: ${NETWORK}"
+    docker network create ${NETWORK}
+  else
+    echo "${NETWORK} already exists"
+  fi
```
Comment (ShishirH) [inline scripts/docker-helpers.sh:146]: Please use tabs instead of spaces

File: scripts/docker-helpers.sh [hunk 2]
```diff
@@ -133,20 +135,29 @@ function docker_prereq() {
...
 }
 
 #
 function docker_setup() {
 	echo "Starting cadvisor container"
-	docker run -d --rm --name=cadvisor   -p ${CADVISOR_PORT}:${CADVISOR_PORT}      --net=host   --cpus=1   --volume=/:/rootfs:ro  --volume=/var/run:/var/run:ro   --volume=/sys:/sys:ro   --volume=/var/lib/docker/:/var/lib/docker:ro   --volume=/dev/disk/:/dev/disk:ro  ${CADVISOR_DOCKER_IMAGE}
+	docker run -d --rm --name=cadvisor   -p ${CADVISOR_PORT}:${CADVISOR_PORT}      --net=${NETWORK}   --cpus=1   --volume=/:/rootfs:ro  --volume=/var/run:/var/run:ro   --volume=/sys:/sys:ro   --volume=/var/lib/docker/:/var/lib/docker:ro   --volume=/dev/disk/:/dev/disk:ro  ${CADVISOR_DOCKER_IMAGE}
```
Comment (ShishirH) [inline scripts/docker-helpers.sh:152]: Can you please eliminate the additional spaces in the command here too?

File: scripts/docker-helpers.sh [hunk 3]
```diff
@@ -133,20 +135,29 @@ function docker_prereq() {
...
 	check_err "Error: cadvisor did not start up"
 
 	echo "Starting prometheus container"
-	docker run -d --rm --name=prometheus -p ${PROMETHEUS_PORT}:${PROMETHEUS_PORT}  --net=host -v ${ROOT_DIR}/${PROMETHEUS_MANIFEST}:/etc/prometheus/prometheus.yml ${PROMETHEUS_DOCKER_IMAGE}
+	docker run -d --rm --name=prometheus -p ${PROMETHEUS_PORT}:${PROMETHEUS_PORT}  --net=${NETWORK} -v ${ROOT_DIR}/${PROMETHEUS_MANIFEST}:/etc/prometheus/prometheus.yml ${PROMETHEUS_DOCKER_IMAGE}
```
Comment (ShishirH) [inline scripts/docker-helpers.sh:156]: Same here.

File: scripts/docker-helpers.sh [hunk 4]
```diff
@@ -133,20 +135,29 @@ function docker_prereq() {
...
 	check_err "Error: prometheus did not start up"
 
 	echo "Starting grafana container"
-	docker run -d --rm --name=grafana    -p ${GRAFANA_PORT}:${GRAFANA_PORT}        --net=host -v ${ROOT_DIR}/${GRAFANA_MANIFESTS}:/etc/grafana/provisioning/ ${GRAFANA_DOCKER_IMAGE}
+	docker run -d --rm --name=grafana    -p ${GRAFANA_PORT}:${GRAFANA_PORT}        --net=${NETWORK} -v ${ROOT_DIR}/${GRAFANA_MANIFESTS}:/etc/grafana/provisioning/ ${GRAFANA_DOCKER_IMAGE}
```
Comment (ShishirH) [inline scripts/docker-helpers.sh:160]: And here.

**Comment by ShishirH** [comment]
Also regarding the commit message, @chandrams can you please verify if this works on Windows?

**Comment by dinogun** [comment]
Fixed up the tabs/spaces issue, @chandrams can open a separate issue if required for windows.

**Pull request merged**